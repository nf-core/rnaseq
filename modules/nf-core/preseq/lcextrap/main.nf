process PRESEQ_LCEXTRAP {
    tag "$meta.id"
    label 'process_single'
    label 'error_ignore'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/preseq:3.1.2--h445547b_2':
        'biocontainers/preseq:3.1.2--h445547b_2' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("*.lc_extrap.txt"), emit: lc_extrap
    tuple val(meta), path("*.log")          , emit: log
    path  "versions.yml"                    , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def paired_end = meta.single_end ? '' : '-pe'
    """
    preseq \\
        lc_extrap \\
        $args \\
        $paired_end \\
        -output ${prefix}.lc_extrap.txt \\
        $bam
    cp .command.err ${prefix}.command.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        preseq: \$(echo \$(preseq 2>&1) | sed 's/^.*Version: //; s/Usage:.*\$//')
    END_VERSIONS
    """
}
