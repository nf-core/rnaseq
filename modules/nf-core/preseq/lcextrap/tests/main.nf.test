nextflow_process {

    name "Test Process PRESEQ_LCEXTRAP"
    script "../main.nf"
    process "PRESEQ_LCEXTRAP"
    tag "modules"
    tag "modules_nfcore"
    tag "preseq"
    tag "preseq/lcextrap"

    test("sarscov2 - single_end") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.modules_testdata_base_path + 'delete_me/preseq/SRR1003759_5M_subset.mr', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.log[0][1]).name).match("single_end - log") },
                { assert snapshot(process.out.lc_extrap).match("single_end - lc_extrap") },
                { assert snapshot(process.out.versions).match("single_end - versions") }
            )
        }
    }

    test("sarscov2 - paired_end") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    [ file(params.modules_testdata_base_path + 'delete_me/preseq/SRR1003759_5M_subset.mr', checkIfExists: true) ]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(file(process.out.log[0][1]).name).match("paired_end - log") },
                { assert snapshot(process.out.lc_extrap).match("paired_end - lc_extrap") },
                { assert snapshot(process.out.versions).match("paired_end - versions") }
            )
        }
    }
}