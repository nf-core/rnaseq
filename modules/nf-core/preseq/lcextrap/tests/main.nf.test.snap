{
    "single_end - lc_extrap": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": true
                    },
                    "test.lc_extrap.txt:md5,1fa5cdd601079329618f61660bee00de"
                ]
            ]
        ],
        "timestamp": "2023-11-23T17:20:40.735535"
    },
    "paired_end - log": {
        "content": [
            "test.command.log"
        ],
        "timestamp": "2023-11-23T17:20:51.981746"
    },
    "single_end - versions": {
        "content": [
            [
                "versions.yml:md5,9a62ff1c212c53573808ccd2137b8922"
            ]
        ],
        "timestamp": "2023-11-23T17:20:40.74601"
    },
    "paired_end - versions": {
        "content": [
            [
                "versions.yml:md5,9a62ff1c212c53573808ccd2137b8922"
            ]
        ],
        "timestamp": "2023-11-23T17:20:52.02843"
    },
    "single_end - log": {
        "content": [
            "test.command.log"
        ],
        "timestamp": "2023-11-23T17:20:40.72985"
    },
    "paired_end - lc_extrap": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.lc_extrap.txt:md5,10e5ea860e87fb6f5dc10f4f20c62040"
                ]
            ]
        ],
        "timestamp": "2023-11-23T17:20:51.998533"
    }
}